Code/tests/test_autopkglib.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+
+import imp
+import plistlib
+import unittest
+from textwrap import dedent
+
+from mock import patch
+
+
+# DO NOT MOVE THIS! This needs to happen BEFORE importing autopkglib
+patch("autopkglib.memoize", lambda x: x).start()
+import autopkglib  # isort: skip
+
+autopkg = imp.load_source("autopkg", "autopkg")
+
+
+class TestAutoPkg(unittest.TestCase):
+    """Test class for AutoPkglib itself."""
+
+    def setUp(self):
+        # This forces autopkglib to accept our patching of memoize
+        imp.reload(autopkglib)
```

Comment:
With the patch created above on line 12, we can now force-reload autopkglib prior to each test being run to reset the memoized values. This allows us to test multiple different outcomes for a memoized function.